### PR TITLE
use vision data to estimate odometry

### PR DESCRIPTION
**Pull request by lmaxwell24**
this would eventually lead into being able to use odometry comfortably in order for estimation when limelight cannot see target

### CODE REVIEW

File: constants.py [hunk 1]
```diff
@@ -184,9 +184,7 @@
 kBackLeftModuleName = "back_left"
 kBackRightModuleName = "back_right"
 
-kLimelightMountingOffset = Translation2d(
-    (kRobotLength / 2) - (4.125 * kMetersPerInch), 0.0
-)
+kLimelightMountingOffset = Translation2d(0.0, 0.0)
```
Comment (LandonBayer) [inline constants.py:187]: I assume this is an implementation thing? Is limelight the center of odometry or something like that?

File: subsystems/visionsubsystem.py [hunk 2]
```diff
@@ -278,11 +278,16 @@ def periodic(self):
...
             SmartDashboard.putBoolean(constants.kTargetPoseArrayKeys.validKey, False)
+            SmartDashboard.putBoolean(
+                constants.kRobotVisionPoseArrayKeys.validKey, False
+            )
         else:
             targetDistance = self.trackingModule.getTargetDistance()
             if targetDistance is None:
                 targetDistance = float("inf")
 
+            targetDistance += constants.kSimTargetUpperHubRadius
```
Comment (LandonBayer) [inline subsystems/visionsubsystem.py:289]: Is this going from hub center? If not, should it be subtracted?

**Review by LandonBayer — APPROVED** [review]
Lots of smartdash but whatever, more benefit than not

File: constants.py (hunk 1 above)
Comment (lmaxwell24) [reply constants.py:187]: limelight is in the middle of the robot

File: subsystems/visionsubsystem.py (hunk 2 above)
Comment (lmaxwell24) [reply subsystems/visionsubsystem.py:289]: the distance to the target's center is the distance to the edge + the radius of the target

**Comment by lmaxwell24** [comment]
verified in lab to work on real robot with expected values

**Review by LandonBayer — APPROVED** [review]
Idk if I approved yet but here you go

**Review by caloisio — APPROVED** [review]

**Pull request merged**